06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * inverted index class
+ * @class
+**/
+class InvertedIndex {
+  /**
+  * class constructor
+  * @constructor
+  **/
+  constructor() {
+    this.fileIndices = {};
+  }
+  /**
+   * Set Index - Sets the indices of all indexed files
+   * @param {String} filename - Name of the indexed file
+   * @param {Object} indices - Indices of the file
+   * @return {Object} Indexed file name and it's indices'
+  **/
+  setIndex(filename, indices) {
+    this.fileIndices[filename] = indices;
+  }
+  /**
+   * Validate File
+   * It checks if a json file is a json array of json objects
+   * @param {Array} file is an array of json objects
+   * @return {Boolean} True if a json file is valid and False otherwise
+  **/
+  static validateFile(file) {
+    if (typeof file !== 'object' || file.length === 0 || !Array.isArray(file)) {
+      return false;
+    }
+    for (let i = 0; i < file.length; i += 1) {
+      const item = file[i];
+      if (!(Object.prototype.hasOwnProperty.call(item, 'title') &&
+      (Object.prototype.hasOwnProperty.call(item, 'text')) &&
+      item.title.length > 0 && item.text.length > 0)) {
+        return false;
+      }
+    }
+    return true;
+  }
+  /**
+    * Tokenize
+    * It splits sentence into an array of refined words
+    * @param {String} text - string of texts
+    * @return {Array} An array of refined split texts
+  **/
+  static tokenize(text) {
+    const remove = /[^\w'\s]|('\s)/g;
+    const lowerCase = text.replace(remove, ' ').toLowerCase();
+    const divideText = lowerCase.split(' ');
+    const sortedText = divideText.sort();
+    return sortedText.filter(item => Boolean(item));
+  }
+  /**
+   * Create Index
+   * It creates the index of a file
+   * @param {String} fileName - Filename of the file to be indexed
+   * @param {Object} fileContent - Content of the uploaded file
+   * @return {Object} An object of each word and their indices in a sorted way
+  **/
+  createIndex(fileName, fileContent) {
+    const indices = {};
+    if (InvertedIndex.validateFile(fileContent)) {
+      fileContent.forEach((doc, docIndex) => {
+        const word = `${doc.title} ${doc.text}`;
+        const tokenized = InvertedIndex.tokenize(word);
+        tokenized.forEach((token) => {
+          if (token in indices) {
+            if (indices[token].indexOf(docIndex) === -1) {
+              indices[token].push(docIndex);
+            }
+          } else {
+            indices[token] = [docIndex];
+          }
+        });
+      });
+      this.setIndex(fileName, indices);
+      return 'Index created';
+    }
+    return 'Index not created';
+  }
+  /**
+   * Get Index
+   * It gets the index of a specified filename
+   * @param {String} filename - Filename of the index to get
```

Comment:
For consistency sake on your code, shouldn't this parameter name be in camelCase?